pkgs/development/libraries/protobuf/protobuf-cmake.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/cmake/install.cmake b/cmake/install.cmake
+index 9b2ae93cf..cef1f5c00 100644
+--- a/cmake/install.cmake
++++ b/cmake/install.cmake
+@@ -137,7 +137,7 @@ install(EXPORT protobuf-targets
+   NAMESPACE protobuf::
+   COMPONENT protobuf-export)
+ 
+-install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_INSTALL_CMAKEDIR}/
++install(DIRECTORY ${CMAKE_INSTALL_CMAKEDIR}/
+   DESTINATION "${CMAKE_INSTALL_CMAKEDIR}"
```

Comment:
I suspect you are trying to deal with an absolute path we set in our default setup-hook:
https://github.com/NixOS/nixpkgs/blob/2ef110ae64a96c0d16df86f8390cfe0af7ff45dc/pkgs/development/tools/build-managers/cmake/setup-hook.sh#L81
But how can this work that you install from a directory in "$out" to itself.
I think, we typically workaround this by doing:
https://github.com/NixOS/nixpkgs/blob/2ef110ae64a96c0d16df86f8390cfe0af7ff45dc/pkgs/development/libraries/libfido2/default.nix#L45

Reply:
Is it preferred to do it through flags instead of patching the (very odd and unfortunate) CMake call?

Reply:
The good thing about "-DCMAKE_INSTALL_LIBDIR=lib" is that we can search for it through nixpkgs and see packages affected.

Also, I am still not sure how your patch works.

Reply:
Looking at the [source code](https://github.com/protocolbuffers/protobuf/blob/b10d490efd6052a02a90277e3325adbec6ce62eb/cmake/install.cmake#L103-L119), it intends to put intermediary file generated by `configure_file` under `lib/cmake/protobuf` in the build directory but (wrongly) assumes the `CMAKE_INSTALL_LIBDIR` is relative.

The line you patch would install the directory from the build directory to the target location but since we have absolute `CMAKE_INSTALL_LIBDIR`, it is already in the target location and not in the build directory (`${CMAKE_CURRENT_BINARY_DIR}`).

The correct patch would be changing the output of `configure_file` and the input of `install(DIRECTORY` to something like `${CMAKE_CURRENT_BINARY_DIR}/this_is_an_intermediary_dir_for_cmake_files`. I do not think there is any need for the intermediate directory to be the same as the target destination.